Add requirements section to the documentation

lesser1 opened the pull request:
I added the first version of a markdown document -- requirements.md -- which includes all of our requirements. I added a menu section to the docs page called "Requirements" analogous to the tooling section.

lesser1 committed:
Add requirements section and initial requirements document

lesser1 committed:
Modify requirements.md
in order to get an initial version

nk-coding commented on docs/requirements/requirements.md:
```diff
@@ -0,0 +1,310 @@
+
+### R-C-001: No external services
+
+- We are not allowed to use external services (e.g., external microservices).
```
I would prefer if we write it from the system's perspective and not from what we are allowed to do.
Imho regarding requirements, the system should be central

lesser1 replied:
I agree. I have changed that, see the latest commit.

nk-coding commented on docs/requirements/requirements.md:
```diff
@@ -0,0 +1,310 @@
+### R-C-002: The use of the system must be free of charge
+
+- Deployment included.
+- Origin: Sarah
```
privacy concern: we do not have consent from all interviewed people
if we put their names publicly on the website, I guess we should at least ask them
imho anonymisation would be an option (just have aliases stakeholder 1, 2, ...)

lesser1 replied:
I agree. I simplified the stakeholders section which now only mentions the different roles (e.g., examiner, supervisor), see the latest commit.

nk-coding commented on docs/requirements/requirements.md:
```diff
@@ -0,0 +1,310 @@
+- A "good" documentation of the system is required.
+- The software architecture must be well documented.
+- When designing the system, we must document our design decisions using so called Architectural Design Records (ADRs).
+  - https://adr.github.io
```
maybe also a link to madr?
https://adr.github.io/madr/

lesser1 replied:
Done.

nk-coding commented on docs/requirements/requirements.md:
```diff
@@ -0,0 +1,310 @@
+
+#### R-SArt-003: Clean Deployment
+
+- The deployment must be "clean".
```
what does this mean? do we have more information here? 
simple deployment was named multiple times, maybe we can add this here

lesser1 replied:
I added this as an open question for us to resolve later.

nk-coding commented on docs/requirements/requirements.md:
```diff
@@ -0,0 +1,310 @@
+
+- Researchers must be able to "simply quickly try something out RE experimentation".
+- It should be really easy and intuitive to quickly deploy the system.
+- The RA should not depend on other systems or services.
```
reference to R-C-001

lesser1 replied:
Done.

nk-coding commented on docs/requirements/requirements.md:
```diff
@@ -0,0 +1,310 @@
+
+#### R-NFR-002: Elasticity
+
+- The system must be *elastic*.
```
maybe some more elaboration

lesser1 replied:
Yes, that would be nice, I added this as a to-do / open question.

lesser1 committed:
Modify requirements.md
in response to PR review done by N. Krieger.
Fix wording issues, rewrite requirements
from the perspective of the system,
add references between requirements,
simplify and anonymize stakeholders section,
note open questions.
